boltdesignsystem/bolt
# Feature/tabs deep linking

## Jira

http://vjira2:8080/browse/BDS-2007

## Summary

Adds deep linking feature to Tabs.

## Details

Users can now link directly to a tab by passing a query string with name "tab" and the value as the tab ID.

This required adding `id` as a Tab Panel prop. Otherwise, IDs are auto-generated and will change on every page load. It is up to users to provide unique IDs. We do not validate.

On load, the page "smooth scrolls" to the linked tab.

## How to test

- Review files changed
- Test the following pages cross-browser.
  - [Tabs deep linking demo](https://feature-tabs-deep-linking.boltdesignsystem.com/pattern-lab/patterns/02-components-tabs-45-tabs-deep-linking/02-components-tabs-45-tabs-deep-linking.html)
  - [Product Page with deep link](https://feature-tabs-deep-linking.boltdesignsystem.com/pattern-lab/patterns/04-pages-10-d8-product-pages-product-t4/04-pages-10-d8-product-pages-product-t4.html?tab=tab-3)
- For the second link, also try refreshing the page, scrolling a bit above or below the linked tab and again refresh. Verify it still scrolls smoothly to the linked tab.
